pkg/controller/networkpolicy/adminnetworkpolicy.go

```diff
@@ -0,0 +1,405 @@
+			antreaServices = append(antreaServices, controlplane.Service{
+				Port: &port,
+			})
+		}
+	}
+	return antreaServices
+}
+
+// splitPolicyPeersByScope splits the ANP/BANP peers in the rule by whether the peer is cluster scoped
+// or per-namespace scoped. Per-namespace peers are those whose defined by sameLabels and
+// notSameLabels.
+func splitPolicyPeerByScope(peers []v1alpha1.AdminNetworkPolicyPeer) ([]v1alpha1.AdminNetworkPolicyPeer, []v1alpha1.AdminNetworkPolicyPeer) {
+	var clusterPeers, perNSLabelPeers []v1alpha1.AdminNetworkPolicyPeer
+	for _, peer := range peers {
+		if peer.Pods != nil && peer.Pods.Namespaces.NamespaceSelector != nil {
+			clusterPeers = append(clusterPeers, peer)
+		} else if peer.Namespaces != nil && peer.Namespaces.NamespaceSelector != nil {
+			clusterPeers = append(clusterPeers, peer)
+		} else {
+			perNSLabelPeers = append(perNSLabelPeers, peer)
+		}
+	}
+	return clusterPeers, perNSLabelPeers
+}
+
+// toAntreaPeerForPolicyCRD processes AdminNetworkPolicyPeers and yield Antrea NetworkPolicyPeers.
+func (n *NetworkPolicyController) toAntreaPeerForPolicyCRD(peers []v1alpha1.AdminNetworkPolicyPeer) (*controlplane.NetworkPolicyPeer, []*antreatypes.AddressGroup) {
+	var addressGroups []*antreatypes.AddressGroup
+	for _, peer := range peers {
+		if peer.Pods != nil {
+			addressGroup := n.createAddressGroup("", &peer.Pods.PodSelector, peer.Pods.Namespaces.NamespaceSelector, nil, nil)
+			addressGroups = append(addressGroups, addressGroup)
+		} else if peer.Namespaces != nil {
+			addressGroup := n.createAddressGroup("", nil, peer.Namespaces.NamespaceSelector, nil, nil)
+			addressGroups = append(addressGroups, addressGroup)
+		}
+	}
+	return &controlplane.NetworkPolicyPeer{
+		AddressGroups: getAddressGroupNames(addressGroups),
+	}, addressGroups
+}
+
+// processClusterSubject processes AdminNetworkPolicySubject and yield Antrea AppliedToGroups.
+func (n *NetworkPolicyController) processClusterSubject(subject v1alpha1.AdminNetworkPolicySubject) []*antreatypes.AppliedToGroup {
+	var appliedToGroups []*antreatypes.AppliedToGroup
+	var atg *antreatypes.AppliedToGroup
+	if subject.Pods != nil {
+		atg = n.createAppliedToGroup("", &subject.Pods.PodSelector, &subject.Pods.NamespaceSelector, nil)
+	} else if subject.Namespaces != nil {
+		atg = n.createAppliedToGroup("", nil, subject.Namespaces, nil)
+	}
+	if atg != nil {
+		appliedToGroups = append(appliedToGroups, atg)
+	}
+	return appliedToGroups
+}
+
+func anpActionToCRDAction(action v1alpha1.AdminNetworkPolicyRuleAction) *antreacrd.RuleAction {
+	actionAllow := antreacrd.RuleActionAllow
+	actionDrop := antreacrd.RuleActionDrop
+	actionPass := antreacrd.RuleActionPass
+	mapping := map[v1alpha1.AdminNetworkPolicyRuleAction]*antreacrd.RuleAction{
+		v1alpha1.AdminNetworkPolicyRuleActionAllow: &actionAllow,
+		v1alpha1.AdminNetworkPolicyRuleActionDeny:  &actionDrop,
+		v1alpha1.AdminNetworkPolicyRuleActionPass:  &actionPass,
+	}
```

Comment:
you can declare a global map to avoid initialiting the map every time.
```
var anpActionToAntreaActionMap = map[v1alpha1.AdminNetworkPolicyRuleAction]antreacrd.RuleAction{
	v1alpha1.AdminNetworkPolicyRuleActionAllow: antreacrd.RuleActionAllow,
	v1alpha1.AdminNetworkPolicyRuleActionDeny:  antreacrd.RuleActionDrop,
	v1alpha1.AdminNetworkPolicyRuleActionPass:  antreacrd.RuleActionPass,
}

func anpActionToAntreaAction(action v1alpha1.AdminNetworkPolicyRuleAction) *antreacrd.RuleAction {
	antreaAction, _ := anpActionToAntreaActionMap[action]
	return &antreaAction
}
```

Reply:
Thanks. Refactored